Automattic/wp-calypso
# Focused Launch: add in-memory router

## Changes proposed in this Pull Request

* Used [`MemoryRouter`](https://reactrouter.com/web/api/MemoryRouter) to handle the routing logic inside the focused launch flow
* Moved the content of the modal into the "summary" route
* Added two additional routes for the domain and plan details

Note: This PR does not contain Typography styles for the router links

## Testing instructions



### Prepare the sandboxed environment:
* Make sure you have an active connection to your sandbox, and that your sandbox is clean and up to date
* In three separate terminal windows, from the root of the project, run:
  - `yarn start`
  - `cd apps/editing-toolkit && yarn dev --sync`
  - `cd apps/wpcom-block-editor && yarn dev --sync`
* Sandbox an _unlaunched_ WordPress site created via the `/start` site creation flow by adding it to your host file

### Test for changes:
* Visit `/page/SANDBOXED_SITE_ID_CREATED_WITH_START.wordpress.com/home?flags=create/focused-launch-flow`
* Click on "Launch" button — Focused Launch modal should appear
* Click on the "View all domains" link — A new placeholder view for the domain details is shown
* Click on "Go back" — the focused launch summary view is shown
* Click on the "View all plans" link — A new placeholder view for the plan details is shown
* Click on "Go back" — the focused launch summary view is shown
* At any point, the "X" button should be visible in top right of the page. Clicking on this button should close the modal. When re-opening the modal, the summary view should always be shown, regardless of the last active view.

For more details about testing instructions, please check this [screen recording](https://d.pr/v/5nBToY)

Fixes #46863
